skrub/dataframe/_pandas.py

```diff
@@ -12,6 +12,52 @@
 from skrub._utils import atleast_1d_or_none
 
 
+def make_dataframe(X, index):
+    """Convert an dictionary of columns into a Pandas dataframe.
+
+    Parameters
+    ----------
+    X : mapping from column name to 1d iterable
+        Input data to convert.
+
+    index : 1d array-like, default=None
+        The index of the dataframe.
+
+    Returns
+    -------
+    X : Pandas dataframe
+        Converted output.
+    """
+    if not isinstance(X, dict) or not all(
```

Comment:
what kind of mistake do you foresee to make this check necessary?

Reply:
It may be slightly overkill, but I think about dictionaries of 2d arrays or dataframes.
For instance:
```python
pd.DataFrame(
	dict(
    	a=[[1, 2, 1]],
	    b=[1, 2, 3],
	)
)
```
Will raise `ValueError: All arrays must be of the same length`, but this error is not informative enough IMO.
I suspect this kind of error might happen more than we'd expect. WDYT?

Reply:
but this is a private function, right? that will be used only by skrub code not directly by users. so if skrub passes it incorrect inputs there isn't too much the user can do about it except report the problem

btw WDYT about renaming `dataframe` to `_dataframe`? we want skrub estimators to support both pandas and polars, but I'm not sure we want to publicly expose the machinery that makes it possible

Reply:
Ok for the renaming, I'll remove the check